### ISSUE TITLE
[ENHANCEMENT] Handle the create/upgrade related work for new harvester CSI driver

### ISSUE THREAD

**Issue by Vicente-Cheng** [harvester/harvester#3467]
**Is your enhancement related to a problem? Please describe.**
The harvester CSI driver ([v0.1.4](https://github.com/harvester/harvester-csi-driver/releases/tag/v0.1.4) and later version) allow the user to specify the StorageClass they wanted. To validate it, we need to add new ClusterRole and ClusterRoleBinding for the RBAC.
There are two cases we need to handle
**Create**
- [x] https://github.com/harvester/harvester/pull/3357

**Upgrade** https://github.com/harvester/harvester/issues/5294
- [x] https://github.com/harvester/harvester/pull/4881

**Describe the solution you'd like**
For v1.1.2, write the document for creating `ClusterRole` and `ClusterRoleBinding` after the upgrade.
For v1.3.0, automatically update the `ClusterRole` and `ClusterRoleBinding` with create/upgrade path

**Additional context**
related issues: https://github.com/harvester/harvester/issues/3052

**Comment by Vicente-Cheng** [comment]
update the document link https://docs.harvesterhci.io/v1.1/rancher/csi-driver#prerequisites-1 for v1.1.2

**Comment by Vicente-Cheng** [comment]
Please do not check the upgrade path.
We have another issue for upgrade https://github.com/harvester/harvester/issues/5294

**Comment by TachunLin** [comment]
Verified **fixed** on `v1.3.0-rc4` with csi-driver `0.1.16`. Close this issue.

### Result
<details><summary>$\color{green}{\textsf{PASS}}$ Check the csi-driver ClusterRole in the RKE2 guest cluster  $~~$</summary>

* On Harvester node, we can find the ClusterRole of the csi-driver with the required content 
  ```
  node1:~ # kubectl get clusterrole harvesterhci.io:csi-driver -o yaml
  apiVersion: rbac.authorization.k8s.io/v1
  kind: ClusterRole
  metadata:
    annotations:
      meta.helm.sh/release-name: harvester
      meta.helm.sh/release-namespace: harvester-system
      objectset.rio.cattle.io/id: default-mcc-harvester-cattle-fleet-local-system
    creationTimestamp: "2024-03-06T07:59:08Z"
    labels:
      app.kubernetes.io/component: apiserver
      app.kubernetes.io/managed-by: Helm
      app.kubernetes.io/name: harvester
      app.kubernetes.io/part-of: harvester
      app.kubernetes.io/version: v1.3.0-rc4
      helm.sh/chart: harvester-1.3.0-rc4
      helm.sh/release: harvester
      objectset.rio.cattle.io/hash: e852fa897f5eae59a44b4bfe186aad80b10b94b3
    name: harvesterhci.io:csi-driver
    resourceVersion: "3774"
    uid: b4ca9bed-b48f-4f75-854f-08d27c88b411
  rules:
  - apiGroups:
    - storage.k8s.io
    resources:
    - storageclasses
    verbs:
    - get
    - list
    - watch
  ```

 </details>

<details><summary>$\color{green}{\textsf{PASS}}$ Check the csi-driver ClusterRoleBinding and Service Account in the RKE2 guest cluster  $~~$</summary>

* On Harvester node, we can find the ClusterRoleBinding and ServiceAccount of the csi-driver with the required content 

  ```
  node1:~ # kubectl get clusterrolebinding -A | grep csi-driver
  default-rke2-v12710                                    ClusterRole/harvesterhci.io:csi-driver                             84m
  node1:~ # kubectl get clusterrolebinding default-rke2-v12710 -o yaml
  apiVersion: rbac.authorization.k8s.io/v1
  kind: ClusterRoleBinding
  metadata:
    creationTimestamp: "2024-03-07T07:30:18Z"
    name: default-rke2-v12710
    ownerReferences:
    - apiVersion: v1
      kind: ServiceAccount
      name: rke2-v12710
      uid: 797617d5-766c-4a1f-ac8d-8424fa34b93a
    resourceVersion: "1399165"
    uid: 95aac075-1f13-46c5-afb4-3c82a6ad2407
  roleRef:
    apiGroup: rbac.authorization.k8s.io
    kind: ClusterRole
    name: harvesterhci.io:csi-driver
  subjects:
  - kind: ServiceAccount
    name: rke2-v12710
    namespace: default
  
  ```

 </details>

### Test Information
* Test Environment: Three nodes Harvester on baremetal machines
* Harvester version: `v1.3.0-rc4`
* Rancher version: `v2.8.2`

### Verify Steps
<details><summary> Check the ClusterRole and ClusterRoleBinding </summary>

  1. Integrate Harvester to Rancher 
  2. Provision a RKE2 guest cluster on Harvester (v1.27.10, v1.25.16)
  3. Access to the guest cluster vm
  4. Export the kubeconfig to run kubectl command
      ```
      export KUBECONFIG=/etc/rancher/rke2/rke2.yaml
      /var/lib/rancher/rke2/bin/kubectl get nodes
      ```
  5. Check the content of the ClusterRole `harvesterhci.io:csi-driver` 
  correspond with https://github.com/harvester/harvester/pull/3357#issue-1528858221
  ```
  $ kubectl get clusterrole harvesterhci.io:csi-driver -o yaml
  ```
  6. Check the content of the ClusterRoleBinding and Service account of   `harvesterhci.io:csi-driver` 
  correspond with https://github.com/harvester/harvester/pull/3357#issue-1528858221
  ```
  $ kubectl get clusterrolebinding -A | grep csi-driver
  $ kubectl get clusterrolebinding <namespace>-<cluster name> -o yaml
  ```

</details>
